src/rgw/rgw_auth.cc

```diff
@@ -1121,6 +1123,22 @@ void rgw::auth::LocalApplier::write_ops_log_entry(rgw_log_entry& entry) const
   }
 }
 
+rgw::auth::LocalApplier::LocalApplier(CephContext* const cct,
+                                      std::unique_ptr<rgw::sal::User>* user,
+                                      std::optional<RGWAccountInfo> account,
+                                      std::vector<IAM::Policy> policies,
+                                      std::string subuser,
+                                      const std::optional<uint32_t>& perm_mask,
+                                      const std::string access_key_id)
+  : user_info(user->get()->get_info()),
+    user(std::move(*user)),
```

Comment:
since you're moving `*user`, the constructor should take `std::unique_ptr<rgw::sal::User>` by value instead of pointer. that requires the caller to pass `std::move(user)`, which makes it obvious at the call site that the `user` variable is being moved from

same goes for the `create_apl_local()` overloads